src/components/ListItem.jsx

```diff
@@ -78,16 +81,52 @@ export function ListItem({ item, listToken }) {
 		}
 	};
 
+	let itemDisplay;
+
+	if (isEditing) {
+		itemDisplay = (
+			<form
+				onSubmit={(e) => {
+					e.preventDefault();
+					setIsEditing(false);
+					try {
+						updateItemName(listToken, item, updatedName);
+					} catch (error) {
+						console.log('error', error);
+					}
+				}}
+			>
+				<input
+					type="text"
+					placeholder="Edit Item"
+					value={updatedName}
+					onChange={(e) => {
+						setUpdatedName(e.target.value);
```

Comment:
There's something we can do here with prevState in order to keep the contents of the input field with the prevState (ie: they won't clear when we click 'edit') but I'm too brain dead to make it work! I've been trying for quite some time. So, if you know how, that would be awesome! Otherwise, something for us to explore on our own another time. 🤣

Reply:
We can make a ternary option in here where value={updatedName ? updatedName : item.name} that way it will fill in with either the updated or the original name? 

I also noticed that if I accidentally submit with nothing in the submit field, it will completely delete everything. Not pressing, a small UI bug that we can fix with an if statement where `if (!updatedName) { updateItemName(listToken, item, item.name} else { updateItemName(listToken, item, updatedName)}` I'm sure there's a better way but that's all my brain is thinking of at the moment (':